Sources/MapboxMaps/Style/Style.swift

```diff
@@ -964,6 +964,38 @@ extension Style {
     }
 }
 
+// MARK: - MapProjection
+
+extension Style {
+    /// Set map projection for Style.
+    /// - Parameter mode: The `MapProjection` to be used by the style.
+    /// - Throws: Errors during encoding or `MapProjectionError.unsupportedProjection` if the supplied projection is not compatible with the SDK.
+    @_spi(Experimental) public func setMapProjection(_ mapProjection: MapProjection) throws {
+        try styleManager.setStyleProjectionForProperties(mapProjection.toJSON())
+    }
+
+    /// Get current map projection for Style.
+    ///
+    /// Please note that even if Style is configured to use `MapProjection.globe`
+    /// starting from `GlobeMapProjection.transitionZoomLevel` and above
+    /// this method will return `MapProjection.mercator`.
+    ///
+    /// - Returns:
+    ///     `MapProjection` map is using.
+    /// - Throws: `MapProjectionError.unsupportedProjection` if the projection is not compatible with the SDK.
+    @_spi(Experimental) public func mapProjection() throws -> MapProjection {
+        let namePropertyName = MapProjection.CodingKeys.name.rawValue
+        let projectionName = styleManager.getStyleProjectionProperty(forProperty: namePropertyName).value as? String
+
+        guard let projection = MapProjection.allCases.first(where: { projectionName == $0.name }) else {
+            throw MapProjectionError.unsupportedProjection
```

Comment:
This is marked as obsolete. Maybe it should just be transferred into Style?

Reply:
There already is a top level `MapProjectionError` in `MapProjection.swift` file, which is used here.